### PR TITLE
base: Make gzip auto-compression skipable

### PR DESCRIPTION
**Pull request by LorbusChris**
Fixes: https://github.com/coreos/fcct/issues/123

If this is agreeable, can we also port it back to v0.2?

### CODE REVIEW

File: base/v0_3_exp/translate_test.go [hunk 1]
```diff
@@ -255,6 +255,7 @@ func TestTranslateFile(t *testing.T) {
 			File{
 				Path: "/foo",
 				Contents: Resource{
+					// String is too short for auto gzip compresison
```
Comment (bgilbert) [inline base/v0_3_exp/translate_test.go:258]:
```suggestion
					// String is too short for auto gzip compression
```

File: base/options.go [hunk 2]
```diff
@@ -15,5 +15,6 @@
 package base
 
 type TranslateOptions struct {
-	FilesDir string // allow embedding local files relative to this directory
+	FilesDir                   string // allow embedding local files relative to this directory
+	DisableGzipAutoCompression bool   // config source compression on Ignition output
```
Comment (bgilbert) [inline base/options.go:19]:
```suggestion
	NoResourceAutoCompression bool   // skip automatic compression of inline/local resources
```

**Comment by bgilbert** [comment]
Yeah, we can add this to 0.2 also.  It's not a user-visible spec change.

**Comment by LorbusChris** [comment]
@bgilbert thanks! I've fixed the typo, renamed the bool and also added it to 0.2.

File: base/v0_2/translate.go [hunk 3]
```diff
@@ -171,7 +171,7 @@ func translateResource(from Resource, options base.TranslateOptions) (to types.R
 	return
 }
 
-func makeDataURL(contents []byte, currentCompression *string) (uri string, gzipped bool, err error) {
+func makeDataURL(contents []byte, currentCompression *string, NoResourceAutoCompression bool) (uri string, gzipped bool, err error) {
```
Comment (arithx) [inline base/v0_2/translate.go:174]: minor: variables should be camelCase

Comment (LorbusChris) [reply base/v0_2/translate.go]: fixed

**Pull request merged**